test/OAuth2AdapterTest.php

```diff
@@ -50,7 +54,9 @@ public function testOAuthServerExceptionRaisedDuringAuthenticateResultsInInvalid
 
         $adapter = new OAuth2Adapter(
             $this->resourceServer->reveal(),
-            $this->response->reveal()
+            function () {
+                return $this->response->reveal();
+            }
```

Comment:
I would add property in that class (and also in other tests), because we have the same usage there many times - and use then just `$this->responseFactory`.